service/refund.go

```diff
@@ -505,20 +506,21 @@ func checkGovPayAndUpdateRefundStatus(req *http.Request, payments []models.Payme
 			return fmt.Errorf("not found error from payment service session with payment ID:[%s]", x.ID)
 		}
 
-		govPayStatusResponse, response, err := service.GovPayService.GetRefundStatus(paymentSession, x.Refunds[0].RefundId)
+		govPayStatusResponse, response, err := service.GovPayService.GetRefundStatus(paymentSession, refund.RefundId)
 
 		if err != nil {
-			log.ErrorR(req, fmt.Errorf("error getting payment pending refund status Id: [%s]", x.Refunds[0].RefundId))
-			return fmt.Errorf("error getting payment pending refund status Id: [%s]", x.Refunds[0].RefundId)
+			log.ErrorR(req, fmt.Errorf("error getting payment pending refund status Id: [%s]", refund.RefundId))
+			return fmt.Errorf("error getting payment pending refund status Id: [%s]", refund.RefundId)
```

Comment:
The value of `err` should be included in the log and returned error

Reply:
Fixed.Thanks